### DISCUSSION TITLE
Mysql datatype : longtext, bigint ....

### DISCUSSION THREAD

**Discussion by SmileJayden** [prisma/prisma#4168] [Q&A] [answered]
When I try to `prisma migrate up --experimental` I want to convert my prisma `String` into mysql `longtext` or `varchar(3000)`, not just `varchar(191)`.

I can not find any hint to specify my mysql`varchar` length in prisma `String` shema. 

Is there any way to specify mysql `varchar` length on prisma shema?
(Like this)
```
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}

model mymodel {
    authors       String<3000>        // here I want to specify longtext type...
}
```

Thanks in advance!

spec:
@prisma/cli: 2.10.0
node : 12.18.4
mysql: 5.7.31

**Comment by ryands17** [comment, marked as answer]
Hey @SmileJayden 👋 
This is currently not possible, you would need to use something like [introspection](https://www.prisma.io/docs/reference/tools-and-interfaces/introspection#overview) for the time being. We will be releasing a new version soon with [native DB types](https://github.com/prisma/prisma/issues/3447) and then you will be able to use longtext and bigint :)

**Reply by SmileJayden** [reply]
Thanks for answer  
Very appreciated :)

**Reply by Dax911** [reply]
Sweet I have been trying to clear up this error for over a day and I have finally found this thread... Any word on that update?

**Reply by janpio** [reply]
What error exactly @Dax911? The discussion here only talks about missing functionality - which has been released for a long time and works perfectly fine as far as we know. If you are using a Prisma 3.x version, you should have not trouble at all with this. If you do, please open a new discussion or issue and provide some additional information. Thanks.

**Reply by Dax911** [reply]
> What error exactly @Dax911? The discussion here only talks about missing functionality - which has been released for a long time and works perfectly fine as far as we know. If you are using a Prisma 3.x version, you should have not trouble at all with this. If you do, please open a new discussion or issue and provide some additional information. Thanks.

Yea sorry I was in a deep rabbit hole there. I just needed to denote that the data was long. I figured it out. My bad. The error said the data was too long I found the documentation for the feature you implemented.

**Reply by Softmatech** [reply]
On my end, I just go to the MySQL table and put the field on **TEXT** type, it works but it will be a little annoying.
<img width="1037" alt="Screen Shot 2022-04-25 at 3 22 56 PM" src="https://user-images.githubusercontent.com/29921959/165159482-85259fb3-23b3-4550-a19e-9142b80e7214.png">

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
